assets/src/edit-story/components/panels/preset/karma/stylePresets.karma.js

```diff
@@ -0,0 +1,238 @@
+          height: 50,
+          backgroundColor: createSolid(1, 15, 15, 0.2),
+          backgroundTextMode: BACKGROUND_TEXT_MODE.HIGHLIGHT,
+        })
+      );
+      const target = fixture.editor.canvas.framesLayer.frame(element.id).node;
+      await selectTarget(target);
+      // Verify that the panel is not found.
+      const addButton = fixture.screen.queryByRole('button', {
+        name: ADD_BUTTON,
+      });
+      expect(addButton).toBeNull();
+    });
+  });
+
+  describe('CUJ: Creator can Apply or Save Text Style from/to Their Preset Library: Save Text Style', () => {
+    beforeEach(async () => {
+      const element = await addText({
+        backgroundColor: createSolid(15, 15, 15, 0.2),
+        backgroundTextMode: BACKGROUND_TEXT_MODE.FILL,
+        content: '<span style="font-style: italic">Hello world</span>',
+        lineHeight: 2,
+      });
+      frame = fixture.editor.canvas.framesLayer.frame(element.id).node;
+    });
+
+    it('should allow adding new text style from a text element', async () => {
+      const addButton = fixture.screen.getByRole('button', {
+        name: ADD_BUTTON,
+      });
+      await addButton.click();
+      const applyPresetButton = fixture.screen.queryByRole('button', {
+        name: APPLY_BUTTON,
+      });
+      // Verify that the preset has been added
+      expect(applyPresetButton).toBeDefined();
+    });
+
+    it('should allow adding new text style from multi-selection', async () => {
+      // Add a second text element & select both.
+      await addText({
+        x: 200,
+        y: 200,
+        width: 250,
+        font: TEXT_ELEMENT_DEFAULT_FONT,
+        content: '<span style="font-weight: 700">Second text</span>',
+      });
+      await selectTarget(frame);
+
+      // Verify that two presets have been added.
+      const addButton = fixture.screen.getByRole('button', {
+        name: ADD_BUTTON,
+      });
+      await addButton.click();
+      const applyPresetButtons = fixture.screen.getAllByRole('button', {
+        name: APPLY_BUTTON,
+      });
+      expect(applyPresetButtons.length).toBe(2);
+      await fixture.snapshot('2 style presets added');
+    });
+  });
+
+  describe('CUJ: Creator can Apply or Save Text Style from/to Their Preset Library: Manage Text Style Presets', () => {
+    // Disable reason: edit icon does not work in test mode for some reason.
+    // eslint-disable-next-line jasmine/no-disabled-tests
+    xit('should allow deleting a text style preset', async () => {
+      // Add text element and style preset.
+      await addText();
+      const addButton = fixture.screen.getByRole('button', {
+        name: ADD_BUTTON,
+      });
+      await addButton.click();
+
+      const applyPresetButton = fixture.screen.queryByRole('button', {
+        name: APPLY_BUTTON,
+      });
+
+      const editButton = fixture.screen.getByRole('button', {
+        name: 'Edit style presets',
+      });
+      fixture.events.click(editButton);
```

Comment:
Here the "Add preset" icon is first click which results in the edit button being rendered, too.
For some reason, however, clicking on the edit-button does not do anything -- when clicking it directly in the browser during the Karma tests running, it also doesn't do anything. Any thoughts on why this might happen? cc @barklund @merapi

Reply:
You're not awaiting the click here, but that's a minor issue.